<launch>

  <!-- Launch World -->
  <include file="$(find ardrone_sim)/launch/world.launch"/>	

  <!-- Spawn simulated quadrotor uav -->
  <include file="$(find ardrone_sim)/launch/spawn_ardrone.launch"/>

  <!-- INCLUDE THE CONTROL NODE -->
  <!-- node pkg="ardro_dyn" type="ardrotest" name="ardrotest" output="screen"/ -->
  
  <!-- node name="ardrone_driver" pkg="ardrone_autonomy" type="ardrone_driver" output="screen"/ -->
  <!-- INCLUDE THE DYNAMIC RECONFIGURATION -->
  <node pkg="rqt_reconfigure" type="rqt_reconfigure" name="rqt_reconfigure" output="screen"/>
  <!-- INCLUDE THE DYNAMIC VIEW -->
  <node pkg="rqt_image_view" type="rqt_image_view" name="rqt_image_view" output="screen"/>
  <!-- INCLUDE TAG DETECTION -->
  <node name="ar_pose" pkg="ar_pose" type="ar_multi" respawn="false" output="screen">
    <param name="marker_pattern" type="string"
      value="$(find ar_pose)/data/object_4x4"/>
    
    <param name="threshold" type="int" value="100"/>
    <param name="use_history" type="bool" value="true"/>
    <remap from="/camera/camera_info" to="/ardrone/camera_info"/>
    <remap from="/camera/image_raw" to="/ardrone/image_raw"/>
    
  </node>	  
  <node name="ar_pose_wall" pkg="ar_pose" type="ar_multi" respawn="false" output="screen">
    <param name="marker_pattern" type="string"
      value="$(find ar_pose)/data/object_4x4"/>
    
    <param name="threshold" type="int" value="100"/>
    <param name="use_history" type="bool" value="true"/>
    <remap from="/camera/camera_info" to="/ardrone/camera_info"/>
    <remap from="/camera/image_raw" to="/ardrone/image_raw"/>
  </node>
</launch>


